File: 0088-merge-sorted-array/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int left = m-1;
        int right = n-1;
        int index = m+n-1;
        while(left >= 0 && right >= 0) {
            if(nums1[left] > nums2[right]) {
                nums1[index] = nums1[left];
                left--;
                index--;
            }
            else {
                nums1[index] = nums2[right];
                right--;
                index--;
            }

        }
        while(left >= 0) {
            nums1[index--] = nums1[left--];
        }
        while(right >= 0) {
            nums1[index--] = nums2[right--];
        }
    }
};

File: 0088-merge-sorted-array/0088-merge-sorted-array.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int sau = m-1;
        int tan = n-1;
        int ans = nums1.length - 1;
        while(sau >=0 && tan >=0){
            if(nums2[tan] >= nums1[sau]){
                nums1[ans] = nums2[tan];
                tan --;
                ans --;
            }
            else {
                nums1[ans] = nums1[sau];
                sau--;
                ans--;
                  }
        }
            while(tan>=0){
                nums1[ans] = nums2[tan] ;
                tan--;      
                ans--;
    }

    }
}
    